substratest/assets.py

```diff
@@ -50,6 +50,50 @@ def get(self):
         return self._asset
 
 
+class ComputePlanFuture(Future):
+    _keys_properties = {
+        'ComputePlan': {
+            'traintuple_keys': 'traintuples',
+            'composite_traintuple_keys': 'composite_traintuples',
+            'aggregatetuple_keys': 'aggregatetuples',
+            'testtuple_keys': 'testtuples',
+        },
+        'ComputePlanCreated': {
+            'traintuple_keys': 'traintuple_keys',
+            'composite_traintuple_keys': 'composite_traintuple_keys',
+            'aggregatetuple_keys': 'aggregatetuple_keys',
+            'testtuple_keys': 'testtuple_keys',
+
+        },
+    }
+
+    def __init__(self, asset, session):
+        self._asset = asset
+        self._getter = session.get_compute_plan
+        for k, v in enumerate(self._keys_properties[asset.__class__.name]):
+            setattr(self, f'_{k}', getattr(asset, v))
+        self._get_traintuple = session.get_traintuple
+        self._get_composite_traintuple = session.get_composite_traintuple
+        self._get_aggregatetuple = session.get_aggregatetuple
+        self._get_testtuple = session.get_testtuple
+
+    def wait(self, timeout=FUTURE_TIMEOUT, raises=True):
+        """wait until all tuples are completed (done or failed)."""
```

Comment:
If `raises=True` it will wait until the first failure.

Reply:
Good catch !

Reply:
I haven't been very clear but the goal of my comment was to update the docstring and not to remove the wait option.

The goal of having a `raises` option is to ensure that by default the test will fail if a tuple execution failed. So I think we should keep it and pass it to the tuple `wait` methods.

Reply:
My reasoning was that we always want a compute plan to wait for all tuples to be over and not raise at the first one that fails. Otherwise we'll have potential pending/executing tuples at the end of the test that may prevent other tests from running successfully.

For example, let's say we have 1 traintuple and 10 composite_tuples, if the traintuple fails, the test would end immediately while there are still 10 ongoing composite tuples.

Reply:
That's indeed a very good reason!